### DISCUSSION TITLE
PCoA using Bray distance with negative values

### DISCUSSION THREAD

**Discussion by hncipriani** [ChiLiubio/microeco_tutorial#18] [Q&A] [answered]
Dears,

After calculating the beta diversity ($cal_betadiv()) with my data normalized by clr ($norm(method = "clr")) I get the following warning:

1: In vegan::vegdist(eco_table, method = i, binary = binary_use, ...) :
  results may be meaningless because data have negative entries
                 in method “bray”

Therefore, theoretically, I should "deal" with these negative values before running a PCoA ($cal_ordination(method = "PCoA")), with "measure= "bray"".

But microeco runs the analysis and plot the graphics without any apparent issue.

I would like to know if microeco apply a correction to deal with the negative values on the data automatically prior to the PCoA analysis.

Yours attentively,
Henrique

**Comment by ChiLiubio** [comment, marked as answer]
Hi. Yes. Bray-Curtis metric needs the count data. It is better to use Bray-Curtis with rarefaction-based normalization. So if you want to use clr-normalized data, it is better to use Eucledian distance, which will be similar with the Aitchison distance (clr + Euclidean). The following is an example:
```
library(microeco)
data(dataset)
d_clr <- trans_norm$new(dataset = dataset)$norm(method = "clr")
d_clr$cal_betadiv("euclidean")
```
Or
```
library(microeco)
data(dataset)
dataset$cal_betadiv("aitchison")
```

Best,
Chi

**Reply by hncipriani** [reply]
Thank you very much for the fast reply, Chi!

Best Regards.
Henrique
